larq_compute_engine/mlir/transforms/translate_tflite.cc

```diff
@@ -0,0 +1,84 @@
+#include "flatbuffers/flexbuffers.h"
+#include "larq_compute_engine/mlir/ir/lce_ops.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "tensorflow/compiler/mlir/lite/ir/tfl_ops.h"
+
+namespace mlir {
+namespace TFL {
+
+namespace {
+
+struct TranslateToLCE : public PassWrapper<TranslateToLCE, FunctionPass> {
+  void getDependentDialects(DialectRegistry& registry) const override {
+    registry.insert<mlir::TFL::TensorFlowLiteDialect>();
+    registry.insert<mlir::lq::LarqDialect>();
+  }
+  void runOnFunction() override;
+};
+
+struct TranslateToLCEPattern : public OpRewritePattern<TFL::CustomOp> {
+  using OpRewritePattern<TFL::CustomOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(TFL::CustomOp custom_op,
+                                PatternRewriter& rewriter) const override {
+    auto stringData = custom_op.custom_option().getValue();
+
+    // Replace CustomOp with relevant LarqOp
+    if (custom_op.custom_code() == "LceQuantize") {
+      rewriter.replaceOpWithNewOp<lq::QuantizeOp>(
+          custom_op, custom_op->getResultTypes(), custom_op->getOperands());
+    } else if (custom_op.custom_code() == "LceDequantize") {
+      rewriter.replaceOpWithNewOp<lq::DequantizeOp>(
+          custom_op, custom_op->getResultTypes(), custom_op->getOperands());
+    } else if (custom_op.custom_code() == "LceBMaxPool2d") {
+      auto map =
+          flexbuffers::GetRoot((uint8_t*)stringData.data(), stringData.size())
+              .AsMap();
+      rewriter.replaceOpWithNewOp<lq::BMaxPool2dOp>(
+          custom_op, custom_op->getResultTypes(), custom_op->getOperands()[0],
```

Comment:
I think this can be simplified a tiny bit:
```suggestion
          custom_op, custom_op->getResultTypes(), custom_op->getOperand(0),
```